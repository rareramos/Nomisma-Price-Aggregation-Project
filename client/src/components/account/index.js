export { Account as default} from './account-container';
