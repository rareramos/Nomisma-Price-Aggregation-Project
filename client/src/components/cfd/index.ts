export { Cfd } from './cfd-container';
