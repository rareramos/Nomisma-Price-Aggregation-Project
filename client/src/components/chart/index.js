export { Chart } from './chart-container';
