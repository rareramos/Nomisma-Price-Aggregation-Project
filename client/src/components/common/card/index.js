export { Card as CenterCard363 } from './card';
