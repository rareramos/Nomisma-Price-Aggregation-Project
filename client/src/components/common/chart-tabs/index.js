import React from 'react';
import {
  arrayOf, func, shape, string,
} from 'prop-types';
import { Nav } from './nav';
import { NavItem } from './nav-item';
import { TabContainer } from './container';

export function ChartTabs(props) {
  const { navItems, selected, setActiveNavItem } = props;
  return (
    <TabContainer>
      <Nav>
        {
          navItems.map((item) => {
            const { title, type } = item;
            return (
              <NavItem
                selected={type === selected}
                key={type}
                onClick={() => setActiveNavItem(type)}
              >
                {title}
              </NavItem>
            );
          })
        }
      </Nav>
    </TabContainer>
  );
}

ChartTabs.propTypes = {
  navItems: arrayOf(shape({ title: string, type: string })).isRequired,
  selected: string.isRequired,
  setActiveNavItem: func.isRequired,
};
