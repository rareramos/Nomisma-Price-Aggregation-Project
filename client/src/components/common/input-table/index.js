import React from 'react';
import {
  arrayOf,
  string,
  oneOfType,
  bool,
  array,
} from 'prop-types';
import { StyledTable } from './styled';

export const InputTable = ({
  header,
  body,
  keys,
}) => (
  <React.Fragment>
    { body.length > 0 && (
      <StyledTable
        header={header}
        body={body}
        keys={keys}
      />
    )}
  </React.Fragment>
);

InputTable.propTypes = {
  header: arrayOf(string).isRequired,
  body: oneOfType([
    arrayOf(array),
    bool,
  ]).isRequired,
  keys: arrayOf(string).isRequired,
};
