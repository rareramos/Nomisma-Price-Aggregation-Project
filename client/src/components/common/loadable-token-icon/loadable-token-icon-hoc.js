/* eslint-disable react/no-multi-comp */
import React from 'react';
import Loadable from 'react-loadable';
import { string } from 'prop-types';
import SvgClose from '@nomisma/nomisma-ui/icons/Close';
import { Loading } from '@nomisma/nomisma-ui/beta/loading';
import memoize from 'lodash/memoize';

export const LoadableTokenIconHOC = memoize(symbol =>
  () => {
    const LoadableIcon = Loadable({
      loader: () =>
        import(`@nomisma/nomisma-ui/token-icons/${symbol[0] + symbol.slice(1).toLowerCase()}.js`),
      render(loaded) {
        const Icon = loaded.default;
        return <Icon />;
      },
      loading: (error) =>
        error ?
          <SvgClose /> :
          <Loading
            width='14px'
            height='14px'
            fill='var(--primary)'
          />,
    });

    return <LoadableIcon />;
  },
);

LoadableTokenIconHOC.propTypes = {
  symbol: string.isRequired,
};
