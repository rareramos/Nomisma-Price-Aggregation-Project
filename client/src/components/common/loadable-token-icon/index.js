import React from 'react';
import { string } from 'prop-types';
import { LoadableTokenIconHOC } from './loadable-token-icon-hoc';

export const LoadableTokenIcon = ({ symbol, ...otherProps}) => {
  const LoadableIcon = LoadableTokenIconHOC(symbol);
  return <LoadableIcon { ...otherProps } />;
};

LoadableTokenIcon.propTypes = {
  symbol: string.isRequired,
};
