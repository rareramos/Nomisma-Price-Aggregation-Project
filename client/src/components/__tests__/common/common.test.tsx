import * as React from 'react';
import { shallow } from 'enzyme';
import { LoadableTokenIcon } from 'components/common/loadable-token-icon';
import { ILoadableTokenIconProps } from 'types/components/common';

const LoadableTokenIconProps : ILoadableTokenIconProps = {
  symbol: 'Abx',
};

it('matches LoadableTokenIcon snapshot', () => {
  const wrapper = shallow(<LoadableTokenIcon {...LoadableTokenIconProps} />);
  expect(wrapper).toMatchSnapshot();
});
