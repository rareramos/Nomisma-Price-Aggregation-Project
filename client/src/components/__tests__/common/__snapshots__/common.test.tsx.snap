// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches LoadableTokenIcon snapshot 1`] = `<Component />`;
